src/libs/ascent/ascent.cpp

```diff
@@ -201,20 +254,115 @@ Ascent::open(const conduit::Node &options)
+                echo_opts.update(msg_ops);
+                if(msg_ops.has_child("echo_threshold") &&
+                   msg_ops["echo_threshold"].dtype().is_string() ) 
+                {
+                    std::string echo_thresh = msg_ops["echo_threshold"].as_string();
+                    if(echo_thresh == "none" || 
+                       echo_thresh == "error" ||
+                       echo_thresh == "warn" )
+                    {
+                        m_verbose_msgs = false;
+                    }
+                    else
+                    {
+                        m_verbose_msgs = true;
+                    }
+                }
+            }
+        }
+        
+        ascent::Logger &logger = ascent::Logger::instance();
+
+        // setup echo
+        logger.set_echo_threshold(echo_opts["echo_threshold"].as_string());
+
+        // control for mpi case
+        // if ranks == "root"
+        //   rank 0 is what is specified, echo_threshold = "none" for all others
+        // if ranks == "all"
+        //   echo_threshold = specified option used for all ranks (alreay handled above)
+        std::string log_ranks_str = echo_opts["ranks"].as_string();
+
+        if(log_ranks_str == "root")
+        {
+            if(par_rank != 0)
+            {
+                logger.set_echo_threshold("none");
+            }
+        }
+
+        // logging options
+        //
+        //   logging: true
+        //
+        //   logging:
+        //      file_pattern: zzzz
+        //      log_threshold:  info
+        //
+
+        Node logging_opts;
+        logging_opts["enabled"] = 0;
+#if defined(ASCENT_MPI_ENABLED)
+        logging_opts["file_pattern"]  = "ascent_log_output_rank_{rank:05d}.yaml";
+#else
+        logging_opts["file_pattern"]  = "ascent_log_output.yaml";
+#endif
+        logging_opts["log_threshold"] = "debug";
+
+        if(m_options.has_path("logging"))
         {
-            std::string msgs_opt = m_options["messages"].as_string();
-            if( msgs_opt == "verbose")
+            if(m_options["logging"].dtype().is_string() &&
+               m_options["logging"].as_string() == "true")
             {
-                m_verbose_msgs = true;
+                logging_opts["enabled"] = 1;
             }
-            else if(msgs_opt == "quiet")
+            else if(m_options["logging"].dtype().is_object())
             {
-                m_verbose_msgs = false;
+                logging_opts["enabled"] = 1;
+                // pull over options
+                logging_opts.update(m_options["logging"]);
             }
         }
 
+        if(logging_opts["enabled"].to_int() == 1)
+        {
+            std::string file_pattern = logging_opts["file_pattern"].as_string();
+        #if defined(ASCENT_MPI_ENABLED)
+            ASCENT_LOG_OPEN( file_pattern ) // serial    
+        #else
+            ASCENT_LOG_OPEN_RANK( file_pattern, par_rank ) // mpi par
+        #endif
```

Comment:
switch? 
```

        #if defined(ASCENT_MPI_ENABLED)
            ASCENT_LOG_OPEN_RANK( file_pattern, par_rank ) // mpi par   
        #else
            ASCENT_LOG_OPEN( file_pattern ) // serial 
        #endif
```

Reply:
thank you for finding this, I am working on tests right now